### ISSUE TITLE
Add per-mod content managers

### ISSUE THREAD

**Issue by Pathoschild** [Pathoschild/SMAPI#257]
Give each mod its own content manager, which should let it...

* load any of the game's content;
* load XNB files from its own folder for its own use;
* _(bonus_) load unpacked `*.png` files from its own folder.

(Split from #173; see also #255.)

**Comment by Pathoschild** [comment]
Split #273 into a separate ticket.

**Comment by Pathoschild** [comment]
Two possible APIs:

* Option 1:
  ```cs
  // load XNB from the mod folder
  helper.Content.Load<Texture2D>("foo.xnb", ContentSource.ModFolder);

  // load unpacked image from the mod folder
  helper.Content.Load<Texture2D>("foo.png", ContentSource.ModFolder);

  // load 'foo' asset from the game's content manager
  helper.Content.Load<Texture2D>("foo", ContentSource.GameContent);
  ```

* Option 2:
  ```cs
  // load XNB from the mod folder
  helper.Content.LoadFromModFolder<Texture2D>("foo.xnb");

  // load unpacked image from the mod folder
  helper.Content.LoadFromModFolder<Texture2D>("foo.png");

  // load 'foo' asset from the game's content manager
  helper.Content.LoadFromGameContent<Texture2D>("foo");
  ```

**Comment by Pathoschild** [comment]
Option 1 is unanimously preferred per discussion on Discord, so I'll go with that.

**Comment by Pathoschild** [comment]
Done in `develop` for the upcoming SMAPI 1.11; pending beta feedback.

**Comment by Pathoschild** [comment]
Released with SMAPI 1.11.
